### ISSUE TITLE
Implement Cookie Consent to comply with GDPR

### ISSUE THREAD

**Issue by Ardnived** [sydevs/WeMeditate#102]
CookieYes seems like a good service and has a free tier which would suit us for the time being.

**Comment by dbackeus** [comment]
What are we using cookies for at the moment?

One approach that eg. Github took was to minimize cookies to a single "essential" cookie (eg. for the Rails session). According to regulation "essential" cookies don't need the whole banner ordeal so that's a pretty sweet option.

**Comment by Ardnived** [comment]
@dbackeus That would definitely be nice if we could. The main one is Google Analytics. I think Anton would consider it essential, but unfortunately the law probably wouldn't agree. 😆 

I think @antontcymbal also added a couple other services. Yandex Analytics and VK for Russia, and I see AdSense in the Tag Manager configuration.

It would be nice if we could do that, but I think given our long term goals (basically the need to improve the site as a funnel to encourage people to local classes), doing without Analytics is probably not an option.

**Comment by Ardnived** [comment]
Of course, if the banner just means that 80% of people disable analytics then there won't be nearly as much value. :/

**Comment by dbackeus** [comment]
Yeah GA and similar external tracking will certainly require consent. I believe there are quite a few open source tracking options (both as in app rubygems or separate applications) available that focus on user privacy that could probably avoid having to deal with extra cookies. But would certainly require time to setup and on boarding for whoever is going to make use of the data.

**Comment by Ardnived** [comment]
@dbackeus Do you know any services like that off the top of your head that you'd recommend? Could be worth taking a look, although I have a feeling Anton will prefer to stick with GA. I've seen that GA has a new beta feature called "Consent Mode" which allows it to continue tracking some things (I'm not sure what) without cookies when the user opts out.

**Comment by dbackeus** [comment]
Here are some recommendations I got:

Rails engines:
ActiveAnalytics - https://github.com/BaseSecrete/active_analytics
Ahoy - https://github.com/ankane/ahoy (this one doesn't come with a built in analytics UI so requires some DIY work)

Open Source self hostable "analytics as a service":
Plausible - https://github.com/plausible/analytics
Posthog - https://github.com/PostHog/posthog

**Comment by Ardnived** [comment]
@antontcymbal Can you take a look at the Google Analytics alternatives that David has mentioned above. To recap this thread, we need to become GDPR compliant, but likely that will mean a significant reduction in analytics data. Using an alternative analytics plugin like the above ones could mean gathering anonymous analytics without the need for GDPR consent.

I'm not sure if this is the right approach, but take a look and let me know what you think. I think [Plausible](https://plausible.io/) looks the most interesting, though maybe the price is more than we would want to spend on analytics as our viewership scales up.

**Comment by Ardnived** [comment]
If we decide to go with a consent manager instead, we should probably go with [Cookiebot](https://www.osano.com/plans). After plenty of research done at work, this is the one we settled on, with built-in support for Google Consent Mode and a pretty cheap plan.

**Comment by Ardnived** [comment]
Here's another good looking option to consider, [Fathom](https://usefathom.com) ($14/mo)

**Comment by antontcymbal** [comment]
Let's use Fathom (usefathom.com). It seems like the easiest and most cost-efficient option. 
I signed up for a 7-day trial, after which we will be billed $14 a month. 
The access is in the 1pass. Will we create a "site" for each language and fire a separate tag in GTM with each site's script based on the URL of the site? "i.e., if it has /it/, fire the script for the italian site. 
In addition, we could also have a script in general, so we could see stats for all the sites at once. 

What do you think?

**Comment by antontcymbal** [comment]
@dbackeus Yes, it is possible, you just have to filter by country, or save a link where the filter is already applied in the parameter, like so: https://app.usefathom.com/share/deasaicp/hilarious+platypus#/?filters=%5B%7B%22property%22%3A%22Country%22%2C%22operator%22%3A%22is%22%2C%22value%22%3A%22United%20States%20of%20America%22%7D%5D&range=last_7_days&site=114903

So i've installed a new tag in GTM with the Fathom script, but for some reason it's not working: GTM adds "" disabling the script. Did I do something wrong, or could it be that GTM doesn't allow competitor's analytics' scripts? See here a screnshot from the preview mode, the tag fires okay, but the script isn't working: https://gyazo.com/ab73e85d3824391547e26e9c7eab5dbf

Also, it seems GTM now has "consent settings", which means we could keep GA analytics and set up a consent pop-up which then enables all the tags which set cookies, without having to pay for or create a cookiebot. GA gives a lot more insights and data, so I'd be tempted to use both GA and Fathom, but only fire GA when we get the consent, thus also optimising the site speed on first load. 
See here a screenshot demonstrating the consent setting in GTA: https://gyazo.com/5c874a957cdc262ac8e59225426c16c7

As for the design of the consent popup, we discussed it with Devindra, it could be neatly on the side, like here: https://firebase.google.com/

Could one of you have a look at why the tag with Fathom script isn't working in GTA, despite firing? And if it can't be set up in GTA, could you just add it to the code directly? @dbackeus @Ardnived

**Comment by Ardnived** [comment]
We'll add it to the code directly, and as David mentioned, I think we definitely want to just have one tag and do filtering within that. I think we should target not needing any consent at all.

Filtering by country is not quite the same as filtering by locale. I'll see if Fathom has a way to include that parameter.

I'll set up Fathom, and then let you know what comes up while doing that.

**Comment by Ardnived** [comment]
Fathom is now implemented and working. Tracking custom events too. Will give it a few weeks before we disable Google Tag Manager.

We may want to brainstorm some more custom events we want to track as well.
